jdaviz/core/loaders/test_loaders.py

```diff
@@ -0,0 +1,67 @@
+import pytest
+import warnings
+from astropy import units as u
+from specutils import SpectralRegion
+from jdaviz.core.registries import loader_resolver_registry
+
+
+def test_loaders_registry(specviz_helper):
+    with pytest.raises(NotImplementedError):
+        specviz_helper.loaders
+    specviz_helper.app.state.dev_loaders = True
+    # built-in loaders: file, url
+    assert len(specviz_helper.loaders) == len(loader_resolver_registry.members)
+
+
+def test_resolver_url(specviz_helper):
+    specviz_helper.app.state.dev_loaders = True
+
+    loader = specviz_helper.loaders['url']
+
+    # no url, no valid formats
+    assert len(loader.format.choices) == 0
+
+    # non-valid input
+    loader.url = 'not-valid-url'
+    assert len(loader.format.choices) == 0
+
+    loader.url = 'https://stsci.box.com/shared/static/exnkul627fcuhy5akf2gswytud5tazmw.fits'  # noqa
```

Comment:
Please separate out the test cases that require remote data to another test function and decorate it with `@pytest.mark.remote_data`.